### PR TITLE
Race: Information about vehicle changes

### PR DESCRIPTION
**Pull request by Nick026**
In race, checkpoints which changes the vehicle now has an icon, and the vehicle name on it. 
A player can enable and disable this in the UI Settings

image

_Reload **mrgreen-settings**  **race_gm** and start **icons-vehicle-change** resource when merging_

### CODE REVIEW

**Comment by Haxardous** [comment]
I think the Design needs to be a little bit tweaked

**Comment by Nick026** [comment]
Of course, what do you have in mind (Already tried centering the icon, but had some trouble so far)

**Comment by Haxardous** [comment]
centering the stuff and changing the font to be boldish would be cool (btw we have a fonts resource)

**Comment by Nick026** [comment]
I'll look into it 👍🏻

**Comment by Nick026** [comment]
image

I've got it centered! Also changed the font (the font resource wasn't working the way I want, so i kept my implementation)

**Comment by Nick026** [comment]
Oh btw, I copied the original icon code from the ghostmode icon.
I realised that the ghostmode icon isn't centered either, should I fix this as well?

**Comment by Haxardous** [comment]
> 
> 
> Oh btw, I copied the original icon code from the ghostmode icon.
> I realised that the ghostmode icon isn't centered either, should I fix this as well?

yeah, btw I think you should make it smaller, really small -- we can discuss that over discord

**Comment by Nick026** [comment]
image

**Review by Haxardous — APPROVED** [review]

File: resources/[gameplay]/icons-vehicle-change/c_client.lua [hunk 1]
```diff
@@ -0,0 +1,119 @@
+local signs = {
+	-- checkpoint { type: String, showSign: boolean, x, y, z: int, guiImage}
+}
+local cpIds = {}
+local enabled = true
+
+local drawDistance = 150
+
+local screenSizex, screenSizey = guiGetScreenSize()
```
Comment (Haxardous) [inline resources/[gameplay]/icons-vehicle-change/c_client.lua:9]: screenSizeX, screenSizeY*

File: resources/[gameplay]/icons-vehicle-change/c_client.lua [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+local guix = screenSizex * 0.1
```
Comment (Haxardous) [inline resources/[gameplay]/icons-vehicle-change/c_client.lua:10]: guiX

File: resources/[gameplay]/icons-vehicle-change/c_client.lua [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+local guiy = screenSizex * 0.1
```
Comment (Haxardous) [inline resources/[gameplay]/icons-vehicle-change/c_client.lua:11]: guiX

File: resources/[gameplay]/icons-vehicle-change/c_client.lua [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+local globalscale = 5
```
Comment (Haxardous) [inline resources/[gameplay]/icons-vehicle-change/c_client.lua:12]: globalScale*

File: resources/[gameplay]/icons-vehicle-change/c_client.lua [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+local globalalpha = .25
```
Comment (Haxardous) [inline resources/[gameplay]/icons-vehicle-change/c_client.lua]: globalAlpha*

File: resources/[gameplay]/icons-vehicle-change/c_client.lua [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+		if(sx) and (sy) then
+			local distanceBetweenPoints = getDistanceBetweenPoints3D(x, y, z, x2, y2, z2)
+			if(distanceBetweenPoints < distance) then
+				dxDrawText(text, sx+2, sy+2, sx, sy, tocolor(R or 255, G or 255, B or 255, alpha or 255 * 0.5), (size or 2.5)-(distanceBetweenPoints / distance), font or "arial", "center", "center")
+			end
+		end
+	end
+end
+
+function enable()
```
Comment (Haxardous) [inline resources/[gameplay]/icons-vehicle-change/c_client.lua:111]: I think the function name should be more clearer.

File: resources/[gameplay]/icons-vehicle-change/c_client.lua [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+	enabled = true
+end
+
+function disable()
```
Comment (Haxardous) [inline resources/[gameplay]/icons-vehicle-change/c_client.lua:115]: I think the function name should be more clearer.

**Pull request merged**